src/test/regress/sql/with.sql

```diff
@@ -1227,3 +1227,42 @@ SELECT i FROM cte a
 JOIN cte b USING (i);
 
 DROP TABLE with_test;
+
+-- Test if sharing is disabled for a SegmentGeneral CTE to avoid deadlock if CTE is
+-- executed with 1-gang and joined with n-gang
+SET optimizer = off;
```

Comment:
Why do we need to disable optimizer here? We run tests on both Postgres optimizer and ORCA, and this test already has two `expected` outputs, we can just update ORCA output even if ORCA test isn't particularly useful. Also, in GPDB 7 this file doesn't have any other `SET optimizer`.